docs/src/guides/architecture-patterns.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Architecture Patterns
+navTitle: Architecture
+description: Design Patterns in Hub.js
+order: 1
+group: 2-concepts
+---
+
+# Architecture Patterns in Hub.js
+
```

Comment:
Spelling error (overs)

I also suggest conveying driving principles rather than historic commentary.

Primary interfaces
Functional implementations 
Extensible, maintainable 

Where should a developer first focus attention and then change based on particular needs.